cmd/clm/main.go

```diff
@@ -115,13 +115,40 @@ func main() {
 
 	execManager := command.NewExecManager(cfg.ConcurrentExternalProcesses)
 
-	p := provisioner.NewClusterpyProvisioner(execManager, clusterTokenSource, secretDecrypter, cfg.AssumedRole, awsConfig, &provisioner.Options{
-		DryRun:          cfg.DryRun,
-		ApplyOnly:       cfg.ApplyOnly,
-		UpdateStrategy:  cfg.UpdateStrategy,
-		RemoveVolumes:   cfg.RemoveVolumes,
-		ManageEtcdStack: cfg.ManageEtcdStack,
-	})
+	provisioners := map[provisioner.ProviderID]provisioner.Provisioner{}
+	for _, provider := range cfg.Providers {
```

Comment:
nit: I think we can consider avoiding the dynamic provider list going forward. The flag `--provider` merely controls which provider is initialized and added to the struct here. The selection of the right provider for each cluster is still being done by the `.Provider` field on the cluster struct.

In other words, what prevents us from unconditionally doing the following?
```go
provisioners[provisioner.ZalandoAWSProvider] = provisioner.NewZalandoAWSProvisioner(...)
provisioners[provisioner.ZalandoEKSProvider] = provisioner.NewZalandoEKSProvisioner(...)
```

Reply:
I agree, no need to loop and switch. Updated.